### ISSUE TITLE
PHP version? (min & max)

### ISSUE THREAD

**Issue by zerai** [scaforchio/LAMPSchool#11]
Sarebbe utile avere esplicitato come requisito (file readme.md o install.txt) la versione PHP (min & max).

Io ho installato una versione locale su PHP5.6 dopo aver trovato questa riga:

```php

$versionephpok = version_compare(substr($versionephp, 0, 3), '5.0', '>=');

```

Non essendoci una suite di test da eseguire e  volendo installare il software su una versione non EOL (7.1 o 7.4) non si ha la certezza che il tutto funzioni a dovere.

Mi sarebbe di aiuto sapere,  prima di apportare modifiche, se qualcuno usa in produzione  lamp-school su PHP7.4 o PHP7.1(che cmq è quasi EOL).

Grazie.

**Comment by mrBackSlash-it** [comment]
Le ultime prove che ho fatto erano OK su php7.4 ma non ho testato tutte le funzioni.

From: zero ***@***.***>
Sent: sabato 9 ottobre 2021 12:57
To: scaforchio/LAMPSchool ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [scaforchio/LAMPSchool] PHP version? (min & max) (#11)


Sarebbe utile avere esplicitato come requisito (file readme.md o install.txt) la versione PHP (min & max).

Io ho installato una versione locale su PHP5.6 dopo aver trovato questa riga:

$versionephpok = version_compare(substr($versionephp, 0, 3), '5.0', '>=');

Non essendoci una suite di test da eseguire e volendo installare il software su una versione non EOL (7.1 o 7.4) non si ha la certezza che il tutto funzioni a dovere.

Mi sarebbe di aiuto sapere, prima di apportare modifiche, se qualcuno usa in produzione lamp-school su PHP7.4 o PHP7.1(che cmq è quasi EOL).

Grazie.

**Comment by zerai** [comment]
@mrBackSlash-it grazie per la risposta. 

Pensandoci bene anche per una 5.6 non vi è certezza finchè qualcuno non dichiara di averla in produzione, o finchè non si scansiona il codice con un qualche tool di compatibilità tra la codebase e una specifica versione PHP.

Speriamo che qualcuno con un sistema in produzione risolva il quesito.

**Comment by mrBackSlash-it** [comment]
Per la 5.6 so che la mia scuola usa quella in produzione, ma è ormai EOL da tanto, quindi si, va scansionato con una suite test.

**Comment by zerai** [comment]
@valerio-bozzolan tra i tuoi contatti (istituti scolastici) sei a conoscenza di sistemi in produzione su versioni php superiori alla 5.6? Se fosse possibile per te reperire questa informazione sarebbe cosa veramente gradita e utile a tutti, chi volesse iniziare e chi ha necessità di aggiornare la propria versione EOL.

Grazie.

**Comment by valerio-bozzolan** [comment]
> sei a conoscenza di sistemi in produzione su versioni php superiori alla 5.6?

Forse volevi chiedere inferiori, giusto? Penso si possa tranquillamente dare per scontato che non ci siano versioni inferiori alla 5.6 sebbene al momento non ho questo tipo di informazione :(

**Comment by zerai** [comment]
No. Intendevo proprio quello che ho scritto, 'superiori alla 5.6', @vittodevit ha già chiarito con la sua risposta che la 5.6 si può usare in produzione (per me non era scontato), ora rimane da capire se è possibile usare lamp-school su versioni superiori alla 5.6. L'installazione (7.4) è possibile, ma non si può avere la certezza sul corretto funzionamento se non usandolo (o testando ogni singola rotta/funzionalità), da qui nasce la domanda che ti ho rivolto. 

Se per ipotesi ci fosse qualcuno che lo usa su versioni 7.0, 7.1, 7.2, che sono comunque versioni EOL è un informazione preziosa, un punto di partenza per chi volesse fare refactoring o aggiungere le proprie personalizzazioni usando i nuovi costrutti del linguaggio (typehint, return type, ecc ecc).

Cmq. nel weekend ho iniziato a scansionare la codebase con dei tool di compatibilità, più avanti posterò un gist o link a repo con i risultati, nel caso interessi a qualcuno.

**Comment by zerai** [comment]
Info sulla compatibilità:

Le versioni 5.4 e 5.6 sono praticamente uguali, vengono riportati gli sessi errori,

Questo è il report per la versione 5.6
![php-56-Screenshot from 2021-10-12 11-20-46](https://user-images.githubusercontent.com/2405279/136929836-7c820df6-e04c-4e40-a671-b47f8db4e5b7.png)

Questo è il report per la scansione che verifica compatibilità da 7.0 a 7.4
![php-70-74-Screenshot from 2021-10-12 11-22-21](https://user-images.githubusercontent.com/2405279/136930034-7a549953-dbbd-4faa-b880-4be1b485e452.png)

Per chi volesse testare con diverse configurazioni [qui c'è il repo](https://github.com/RegitSchool/ls-analisi-compabilita-php).  (repository di analisi, non di sviluppo come scritto nel readme)

Quando mi sarà possibile aggiungerò una ulterire analisi per quel che riguarda OSWAP e criticità varie.
